backend/core/src/main/java/org/sonarsource/sonarlint/core/repository/reporting/PreviouslyRaisedFindingsRepository.java

```diff
@@ -84,4 +84,10 @@ public Optional<RaisedHotspotDto> getRaisedHotspotWithScopeAndId(String scopeId,
       .findFirst();
   }
 
+  public Optional<RaisedIssue> findRaisedIssueById(String configurationScopeId, UUID issueId) {
+    return Optional.ofNullable(previouslyRaisedIssuesByScopeId.get(configurationScopeId))
+      .flatMap(issuesByUri -> issuesByUri.entrySet().stream()
+        .flatMap(entry -> entry.getValue().stream().filter(issue -> issue.getId().equals(issueId)).findFirst().map(issue -> new RaisedIssue(entry.getKey(), issue)).stream())
+        .findFirst());
```

Comment:
```suggestion
return Optional.ofNullable(previouslyRaisedIssuesByScopeId.get(configurationScopeId))
    .flatMap(issuesByUri -> issuesByUri.entrySet().stream()
        .flatMap(entry -> entry.getValue().stream()
            .filter(issue -> issue.getId().equals(issueId))
            .map(issue -> new RaisedIssue(entry.getKey(), issue)))
        .findFirst());
```
We could use better indentation for readability, and I think we could eliminate the first `.findFirst()` and `.stream()` for simplicity?